### ISSUE TITLE
Very complex configuration

### ISSUE THREAD

**Issue by fatalistt** [skoruba/IdentityServer4.Admin#430]
In `Startup.ConfigureServices` of Admin and AdminApi projects there are lots of Type parameters (_[Admin, for example](https://github.com/skoruba/IdentityServer4.Admin/blob/119b1f611ad956be50b834c19b6754e1954dd016/src/Skoruba.IdentityServer4.Admin/Startup.cs#L49-L66)_).
It's very difficult to understand, which parameter user should to change if he want to change something.
I suppose to create a builder to split this method. It'll be looks something like that:
```
services.AddAdminAspNetIdentityServices(); // With default values like string keys, DbContexts, etc.
services.AddAdminAspNetIdentityServices(admin => {
    admin.ConfigureIdentity(identity => {
        identity.ConfigureKeys(keys => {
            keys.UseUserKey<string>();
            ...
        })
        .ConfigureEntities(entities => {
            entities.UseUser<IdentityUser<string>>();
            ...
        });
    }
    ...
});
```
In configuration we start from default values, so user can skip some blocks like `ConfigureKeys`.
I start to making this, but I have some question for now (may be I'll add some more later):
In `Skoruba.IdentityServer4.Admin.EntityFramework.Identity.Repositories.IdentityRepository` you have `TUserKey`, `TRoleKey`, and `TKey` parameters ([src](https://github.com/skoruba/IdentityServer4.Admin/blob/119b1f611ad956be50b834c19b6754e1954dd016/src/Skoruba.IdentityServer4.Admin.EntityFramework.Identity/Repositories/IdentityRepository.cs#L19)).
`TUserKey` and `TRoleKey` are never used except these 2 methods: [`ConvertUserKeyFromString`](https://github.com/skoruba/IdentityServer4.Admin/blob/119b1f611ad956be50b834c19b6754e1954dd016/src/Skoruba.IdentityServer4.Admin.EntityFramework.Identity/Repositories/IdentityRepository.cs#L49-L56) and [`ConvertRoleKeyFromString`](https://github.com/skoruba/IdentityServer4.Admin/blob/119b1f611ad956be50b834c19b6754e1954dd016/src/Skoruba.IdentityServer4.Admin.EntityFramework.Identity/Repositories/IdentityRepository.cs#L58-L65)
All other parameters are depend from `TKey` (`TUser : IdentityUser<TKey>`, for example).
Later you use `UserManager<IdentityUser<TKey>>.Users.AnyAsync(x => x.Id.Equals(_instance of TUserKey_));` [here](https://github.com/skoruba/IdentityServer4.Admin/blob/119b1f611ad956be50b834c19b6754e1954dd016/src/Skoruba.IdentityServer4.Admin.EntityFramework.Identity/Repositories/IdentityRepository.cs#L67-L72).
If `TKey` and `TUserKey` will be a different types, this code will fall, isn't it?

**Comment by skoruba** [comment]
You are right, this configuration is maybe too complex. 
Yes, `TKey` and `TUserKey` must be same type. I like your approach, we can rewrite this, if you can send some PR, it would be great.

**Comment by fatalistt** [comment]
Yes, I will make a PR later when i finished it.
If `TKey` and `TUserKey` (and `TRoleKey`) must be the same type, why do we need `TUserKey` and `TRoleKey` ever?

**Comment by skoruba** [comment]
This is great question, I would prefer separate keys for User and Role, maybe we can remove `TKey` at all and keep only `TUserKey` and `TRoleKey`. What do you think?

**Comment by fatalistt** [comment]
Yes, it's better to remove `TKey`, but what about the keys in `TUserClaim`, `TUserRole`, `TUserLogin`, `TRoleClaim` and `TUserToken`? Which keys they should to use? I didn't work with ASP.Net Identity so deep before, so I know nothing about these keys.

**Comment by skoruba** [comment]
These objects are connected with `TUserKey` and `TRoleKey`. For example: `TUserClaim` is object which uses `TUserKey`. You are not able to specify primary key for `TUserClaim` but you can specify primary key for User which is used in `TUserClaim`. Does it make sense for you?

**Comment by fatalistt** [comment]
Yes, I think, it does. But after looking to sources of Identity there are some questions.
`IdentityUserClaim<TUserKey>`, `IdentityUserLogin<TUserKey>`, `IdentityRoleClaim<TRoleKey>` and `IdentityUserToken<TUserKey>` are contains only one of [`TUserKey UserId`, `TRoleKey RoleId`].
But `IdentityUserClaim<TKey>` contains both `TKey RoleId` and `TKey UserId`, which means that `TUserKey` and `TRoleKey` are the same type `TKey`, so we're not able to split `TKey` to two different types. Am i right?

**Comment by skoruba** [comment]
Can you send me which part do you mean?

**Comment by fatalistt** [comment]
Yes, of course.
[IdentityRoleClaim<TKey>](https://github.com/aspnet/AspNetCore/blob/3b7cdc166aa3c9733fa60b56d91fc0fff9b11652/src/Identity/Extensions.Stores/src/IdentityUserRole.cs#L12)

**Comment by skoruba** [comment]
You are right, thanks for this point. We should use only one key - TKey, becuase of this limitation.

**Comment by b0** [comment]
Hi @skoruba and @fatalistt,

I was looking at this issue and associated PR and me too find that configuration is a bit complex and definitively there is TRoleKey and TUserKey type params too much.

I find PR maybe a bit too big so maybe it's possible to split it in two, one for TRoleKey and TUserKey removal and another for simplified configuration?

What do you think? 

Thank you

**Comment by skoruba** [comment]
Hi @b0 I definitely agree with you, I am happy for this PR, but it is too complex - splitting would be perfect.
